tasks/update_go.py

```diff
@@ -0,0 +1,171 @@
+import hashlib
+import re
+import sys
+from typing import Dict, Optional, Set, Tuple, Type
+
+import requests
+from invoke import Context, exceptions, task
+
+# a platform is an OS and an architecture
+Platform = Type[Tuple[str, str]]
+
+# list all platforms used in the repo
+# store the platform names in the format expected by Go
+# the variables in the Dockerfiles should be stored in uppercased variables
+# see https://go.dev/dl/ for the full list of platforms
+PLATFORMS: Set[Platform] = {
+    ("linux", "amd64"),
+    ("linux", "arm64"),
+    ("linux", "armv6l"),
+    ("windows", "amd64"),
+}
+
+# list of Dockerfiles where we want to replace Go version and sha variables
+# hardcode the number of expected matches so that we can warn if we get something else
+DOCKERFILES: Dict[str, int] = {
+    "./circleci/Dockerfile": 2,
+    "./deb-arm/Dockerfile": 3,
+    "./deb-x64/Dockerfile": 1,
+    "./rpm-arm64/Dockerfile": 2,
+    "./rpm-armhf/Dockerfile": 2,
+    "./rpm-x64/Dockerfile": 2,
+    "./suse-x64/Dockerfile": 2,
+    "./system-probe_arm64/Dockerfile": 2,
+    "./system-probe_x64/Dockerfile": 2,
+}
+
+
+# returns a map from a pattern to what it should be replaced with, for dockerfiles
+def _get_dockerfile_patterns(
+    version: str, shas: Dict[Platform, str]
+) -> Dict[re.Pattern, str]:
+    patterns: Dict[re.Pattern, str] = {
+        re.compile(
+            "^(ARG GO_VERSION=)[.0-9]+$", flags=re.MULTILINE
+        ): rf"\g<1>{version}",
+    }
+    for (os, arch), sha in shas.items():
+        varname = f"GO_SHA256_{os.upper()}_{arch.upper()}"
+        pattern = re.compile(f"^(ARG {varname}=)[a-z0-9]+$", flags=re.MULTILINE)
+        replace = rf"\g<1>{sha}"
+
+        patterns[pattern] = replace
+
+    return patterns
+
+
+# returns a map from a pattern to what it should be replaced with, for windows file
+def _get_windows_patterns(
```

Comment:
could this method and the previous one share their code? There are very few differences.

Reply:
They were similar because I copy pasted the first to write the second :smile:
I simplified `_get_windows_pattern` to just return the patterns really used in the windows file